k-distribution/tests/regression-new/set-symbolic-tests/failing/intersect-37-spec.k.out.actual

```diff
@@ -0,0 +1,10 @@
+  <k>
+    intersectResult ( false ) ~> .
+  </k>
+#And
+  {
+    false
+  #Equals
+    X:MyId in SetItem ( x )
```

Comment:
Absolutely no idea what happens here! The prover appears to be inventing the `X /= x` constraint, maybe by way of confusing `x` and `y`?
Without the new rules, the result is as expected:
```
  #Not ( {
    X
  #Equals
    y
  } )
#And
  <k>
    intersectResult ( X:MyId in intersectSet ( SetItem ( X:MyId )
    SetItem ( y ) , SetItem ( x ) ) ) ~> .
  </k>
```

Reply:
As it turned out, the rule `rule (S SetItem(X)) |Set S  => S SetItem(X)` is causing this confusion (or at least removing it makes this test behave again). I don't quite understand why.

Reply:
Please open an issue for this.

Reply:
https://github.com/runtimeverification/haskell-backend/issues/3294